kimchi/src/verifier_index.rs

```diff
@@ -248,6 +258,58 @@ impl<G: CommitmentCurve> VerifierIndex<G>
 where
     G::BaseField: PrimeField,
 {
+    ///
+    // TODO: this is highly error-prone...
+    pub fn absorb<EFqSponge>(&self, sponge: &mut EFqSponge)
+    where
+        EFqSponge: FqSponge<G::BaseField, G, G::ScalarField>,
+    {
+        // absorb all the commitments
+        let commitments = self
+            .sigma_comm
+            .iter()
+            .chain(&self.coefficients_comm)
+            .chain([&self.generic_comm])
+            .chain([&self.psm_comm])
+            .chain([&self.complete_add_comm])
+            .chain([&self.mul_comm])
+            .chain([&self.emul_comm])
+            .chain([&self.endomul_scalar_comm])
+            .chain(self.chacha_comm.iter().flatten())
+            .chain(&self.range_check_comm);
+
+        for commitment in commitments {
+            sponge.absorb_g(&commitment.unshifted);
```

Comment:
I think (for pickles reasons) we want to hash the verifier index and then absorb that hash. Can you change this to do that?